src/Hooks/UsePosts.js

```diff
@@ -18,5 +18,8 @@ export const UsePosts = (userId = null) => {
     fetchPosts();
   }, [userId]);
 
-  return posts;
+  return {
+    posts,
+    setPosts,
+  }
```

Comment:
postsとsetPostsを辞書型で返すようにしました。
これによってpostsの書き換えが行えます。

さらにこのhooksにロジックをまとめていく場合は
```suggestion
const updatePost = (editedPost) => {

    setPosts((prevPosts) => {
      return prevPosts.map((post) => {
        if (post.id === editedPost.id) {
          return { ...post, ...editedPost };
        } else {
          return post;
        }
      });
    });
}
...

return {
 posts,
  updatePost,
  }
```
のようにできます。同様の手法でgoodCountやremoveCountも同様にこのhooksに集められます。
この手法ですと、hooksの中にpost変更関連を全て集められるのでHome.js側がスッキリします。